tests/roles/control_plane_rollback/tasks/main.yaml

```diff
@@ -0,0 +1,149 @@
+        for i in {1..3}; do
+            SSH_CMD=CONTROLLER${i}_SSH
+            if [ ! -z "${!SSH_CMD}" ]; then
+                if ${!SSH_CMD} sudo systemctl is-enabled $service &> /dev/null; then
+                    if ! ${!SSH_CMD} systemctl show $service | grep ActiveState=active >/dev/null; then
+                        echo "ERROR: Service $service is not running on controller $i"
+                    else
+                        echo "OK: Service $service is running in controller $i"
+                    fi
+                fi
+            fi
+        done
+    done
+
+    echo "Starting pacemaker OpenStack services"
+    for i in {1..3}; do
+        SSH_CMD=CONTROLLER${i}_SSH
+        if [ ! -z "${!SSH_CMD}" ]; then
+            echo "Using controller $i to run pacemaker commands"
+            for resource in ${PacemakerResourcesToStart[*]}; do
+                if ${!SSH_CMD} sudo pcs resource config $resource &>/dev/null; then
+                    echo "Starting $resource"
+                    ${!SSH_CMD} sudo pcs resource enable $resource
+                else
+                    echo "Service $resource not present"
+                fi
+            done
+            break
+        fi
+    done
+
+    echo "Checking pacemaker OpenStack services"
+    for i in {1..3}; do
+        SSH_CMD=CONTROLLER${i}_SSH
+        if [ ! -z "${!SSH_CMD}" ]; then
+            echo "Using controller $i to run pacemaker commands"
+            for resource in ${PacemakerResourcesToStop[*]}; do
+                if ${!SSH_CMD} sudo pcs resource config $resource &>/dev/null; then
+                    if ${!SSH_CMD} sudo pcs resource status $resource | grep Started >/dev/null; then
+                        echo "OK: Service $resource is started"
+                    else
+                        echo "ERROR: Service $resource is stopped"
+                    fi
+                fi
+            done
+            break
+        fi
+    done
+
+- name: clean up the podified control plane
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc delete --ignore-not-found=true --wait=false openstackcontrolplane/openstack
+    oc patch openstackcontrolplane openstack --type=merge --patch '
+    metadata:
+      finalizers: []
+    ' || true
+
+    while oc get pod | grep rabbitmq-server-0; do
+        sleep 2
+    done
+    while oc get pod | grep openstack-galera-0; do
+        sleep 2
+    done
+
+    oc delete --ignore-not-found=true --wait=false pod mariadb-copy-data
+    oc delete --ignore-not-found=true --wait=false pvc mariadb-data
+    oc delete --ignore-not-found=true --wait=false pod ovn-copy-data
+    oc delete --ignore-not-found=true secret osp-secret
+
+- name: verify that the source control plane is responding
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    ssh -i {{ edpm_privatekey_path }} root@{{ edpm_node_ip }} OS_CLOUD=standalone openstack user list
```

Comment:
Can we make this command more generic. So that, it will verify the controlplane in case of HA OSP cloud deployed on RHV or other platform instead of standalone specific.

Reply:
And i tweaked it a bit more in https://github.com/openstack-k8s-operators/data-plane-adoption/pull/372/commits/10522966d5297e2ee4dca9302a763014fdcde3dc to make it use `$CONTROLLER1_SSH`. But still there's the `OS_CLOUD=standalone` bit so the variable may need overriding in non-standalone environments anyway.

Reply:
The override variable can be put in standard vars.yaml?

Reply:
Yes i think that should work.